jazz_create-serverless-service/test/test.js

```diff
@@ -22,234 +22,667 @@ const awsContext = require('aws-lambda-mock-context');
+        }
+      };
+      let responseObject_update = {
+        statusCode: 200,
+        body: {
+          data: "Service catalog updated"
+        }
+      };
+      event.stage = "dev";
+      let config = configObj(event);
+// wrapping requests 
+      reqStub = sinon.stub(request, "Request", (obj) => {
+ // Matching response Object to the corresponding Request call
+        if (obj.uri === (config.SERVICE_API_URL + config.TOKEN_URL)) {
+          return obj.callback(null, responseObject_getToken, responseObject_getToken.body);
+        } else if (obj.uri === "https://{conf-apikey}.execute-api.{conf-region}.amazonaws.com/dev/jazz/services") {
+          return obj.callback(null, responseObject_createService, responseObject_createService.body);
+        } else if (obj.url === '{conf-jenkins-host}/job/create-service/buildWithParameters') {
+          let errObject = {
+            message: responseObject_serviceOnboarding.body.message,
+            jenkins_api_failure: true
+          }
+          return obj.callback(errObject, responseObject_serviceOnboarding, responseObject_serviceOnboarding.body);
+        } else if (obj.uri = 'https://{conf-apikey}.execute-api.{conf-region}.amazonaws.com/dev/jazz/services/ghd93-3240-2343') {
+          obj.callback(null, responseObject_update, responseObject_update.body);
+        }
+      });
+      let callFunction = index.handler(event, context, (err, res) => {
+        err = JSON.parse(err);
+        if (err.message == "Service onboarding jenkings build Failed") {
+          bool = true;
+        }
+        assert.isTrue(bool);
+      })
+    })
+  })
+  describe("getToken", () => {
+    let config, event;
+    beforeEach(() => {
+      event = {
+        "stage": "test",
+        "headers": {
+          "Authorization": "fr1end$hip_1s_mAg1c"
+        },
+        "principalId": "@pp1eJack",
+        "body": {
+          "service_name": "test-service",
+          "service_type": "function",
+          "domain": "test-domain",
+          "runtime": "nodejs",
+          "approvers": ['tw1light_$pArkle'],
+          "rateExpression": "1 * * * ? *",
+          "slack_channel": "mlp_fim",
+          "require_internal_access": false,
+          "create_cloudfront_url": false //, //?
+          //"enableEventSchedule" : false
+        }
+      };
+      config = configObj(event);
+    })
+    it("Should Return authToken when called with valid paramenters", () => {
+      let bool = false;
+      let responseObject = {
+        statusCode: 200,
+        body: {
+          data: {
+            "token": "ghd93-3240-2343"
+          }
+        }
+      };
+      let reqStub = sinon.stub(request, "Request", (obj) => {
+        return obj.callback(null, responseObject, responseObject.body);
+      });
+      index.getToken(config).then((res) => {
+        if (res && res === "ghd93-3240-2343") {
+          bool = true;
+        }
+        assert.isTrue(bool);
+      })
+      reqStub.restore();
```

Comment:
normally when you stub you should also verify sinon.assert.calledOnce(or twice) on the stub to ensure it is indeed being called.